src/neml2/dispatchers/SimpleMPIScheduler.cxx

```diff
@@ -0,0 +1,165 @@
+// Copyright 2024, UChicago Argonne, LLC
+// All Rights Reserved
+// Software Name: NEML2 -- the New Engineering material Model Library, version 2
+// By: Argonne National Laboratory
+// OPEN SOURCE LICENSE (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "neml2/dispatchers/SimpleMPIScheduler.h"
+#include "neml2/misc/assertions.h"
+
+#include "hwloc.h"
+
+#include <string>
+#include <functional>
+
+namespace neml2
+{
+
+register_NEML2_object(SimpleMPIScheduler);
+
+OptionSet
+SimpleMPIScheduler::expected_options()
+{
+  OptionSet options = WorkScheduler::expected_options();
+  options.doc() =
+      "Dispatch work to a single device selected based on processor ID in given batch sizes.";
+
+  options.set<std::vector<Device>>("devices");
+  options.set("devices").doc() = "List of devices to dispatch work to";
+
+  options.set<std::size_t>("batch_size");
+  options.set("batch_size").doc() = "Batch size";
```

Comment:
Should we allow different batch sizes for devices?

Reply:
Yep